src/test/java/org/ehrbase/fhirbridge/fhir/procedure/GenericTherapyIT.java

```diff
@@ -0,0 +1,157 @@
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.GeccoProzedurKategorieElement;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.NichtDurchgefuehrteProzedurEvaluation;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.ProzedurAction;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.UnbekannteProzedurEvaluation;
+import org.ehrbase.fhirbridge.fhir.AbstractMappingTestSetupIT;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Procedure;
+import org.javers.core.Javers;
+import org.javers.core.JaversBuilder;
+import org.javers.core.diff.Diff;
+import org.javers.core.metamodel.clazz.ValueObjectDefinition;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class GenericTherapyIT extends AbstractMappingTestSetupIT {
+
+    public GenericTherapyIT() {
+        super("Procedure/Therapy/", Procedure.class);
+    }
+
+    @Test
+    void createApheresisProcedure() throws IOException {
+        create("apheresis-example-1.json");
+        create("apheresis-example-2.json");
+        create("apheresis-example-3.json");
+        create("apheresis-example-4.json");
+        create("apheresis-example-5.json");
+    }
+
+    @Test
+    void createDialysisProcedure() throws IOException {
+        create("dialysis-example-1.json");
+        create("dialysis-example-2.json");
+        create("dialysis-example-3.json");
+        create("dialysis-example-4.json");
+        create("dialysis-example-5.json");
+        create("dialysis-example-6.json");
+    }
+
+    @Test
+    void createExtracorporealMembraneOxygenation() throws IOException {
+        create("extracorporeal-membrane-oxygenation-example-1.json");
+        create("extracorporeal-membrane-oxygenation-example-2.json");
+        create("extracorporeal-membrane-oxygenation-example-3.json");
+        create("extracorporeal-membrane-oxygenation-example-4.json");
+    }
+
+    @Test
+    void createPronePosition() throws IOException {
+        create("prone-position-example-1.json");
+        create("prone-position-example-2.json");
+        create("prone-position-example-3.json");
+        create("prone-position-example-4.json");
+    }
+
+    @Test
+    void createRadiologyProcedures() throws IOException {
+        create("radiology-example-1.json");
+        create("radiology-example-2.json");
+        create("radiology-example-3.json");
+        create("radiology-example-4.json");
+    }
+
+    @Test
+    void createRespiratoryTherapies() throws IOException {
+        //create("respiratory-therapies-example-1.json");
+        //create("respiratory-therapies-example-2.json");
+        //create("respiratory-therapies-example-3.json"); // INVALID NAME
```

Comment:
why are these examples commented ?

Reply:
These examples threw an error because the opt was missing some terminologies. It is fixed now